src/blocks/HeadingBlock/HeadingBlock.astro

```diff
@@ -0,0 +1,23 @@
+---
+import type { HeadingBlockFragment } from '@lib/datocms/types';
+
+export interface Props {
+  block: HeadingBlockFragment
+}
+
+const { block } = Astro.props;
+
+const Element = `h${block.level || 2}`;
+const hasSubHeading = Boolean(block.subHeading);
+---
+
+{hasSubHeading === false &&
+  <Element>{block.heading}</Element>
+}
+
+{hasSubHeading === true &&
+  <Element>
+    {block.heading}<br />
+    <small>{block.subHeading}</small>
+  </Element>
+}
```

Comment:
Consider making this a ternary statement for legibility: (also including those earlier comments)
```suggestion
{subtitle
  ? <Tag>
      {title}<br />
      <small>{subtitle}</small>
    </Tag>
  : <Tag>{title}</Tag>
}
```